lib/src/client/channel.dart

```diff
@@ -22,6 +22,9 @@ import 'call.dart';
 import 'connection.dart';
 import 'method.dart';
 
+/// Handler to be called when the channel is shut down.
+typedef ChannelShutdownHandler = void Function();
```

Comment:
Do we need such a simple `typedef`?

Reply:
No, it's a bad API.

It's also breaking for google3 to remove it.

Reply:
I can give a shot at cleanup and see how long it takes to drop it.

Reply:
It looks relatively easy to remove internally, I'll send out a CL and hopefully the owners of the code with usage will quickly approve.

If not, I may also be able to make the typedef internal only with a patch file.

Reply:
I was able to remove the typedef internally.